autotools: revert '-std=gnu89' (commit f310326)

claudioandre-br opened the pull request:
Apparently commit e2e868d fixed the remaining issues with gcc 5.x. See also #1093, https://github.com/openwall/john/commit/e2e868db3e153b3f959e119a51703d4afb99c624, and https://github.com/openwall/john/commit/f310326a79dc869b0db1329675b1cd072798e436.

Reconfigured using a:
```
docker run --rm -v $(pwd):/host -w /host ubuntu:18.04 bash -c \
  "apt update && apt install -y autoconf sudo; autoconf --version && cd src && useradd -U -m JtR; sudo -u JtR autoreconf -i"

```

ghost commented:
Windows CI is Ok. This CI is also happy (it has gcc  5.x and, e.g., gcc 10). 

Closes #5324.

solardiz commented:
Why is the move of the `CFLAGS="$CFLAGS $SIMD_FLAGS -O0"` line seen in the diff here - haven't we done it before? Have we not regenerated the `configure` file then? If so, I think this should be a separate commit now (first commit in this PR, then make the actual new change with a second commit).

> Apparently commit e2e868db3e153b3f959e119a51703d4afb99c624 fixed the remaining issues with gcc 5.x.

That commit looks really weird/wrong. If we still have those changes/hacks, we'll need to revisit them - but can do so separately from this PR.

Commit:
autotools: regenerate the configure file

Commit:
autotools: revert '-std=gnu89' (commit f310326)

Apparently commit e2e868d fixed the remaining issues with gcc 5.x.
See also #1093, https://github.com/openwall/john/commit/e2e868db3e153b3f959e119a51703d4afb99c624,
and https://github.com/openwall/john/commit/f310326a79dc869b0db1329675b1cd072798e436.
